docs/type-definitions/inputs.md

```diff
@@ -108,3 +109,48 @@ $variables = [
 
 **graphql-php** will validate the input against your InputObjectType definition and pass it to your
 resolver as `$args['filters']`
+
+## Converting input object array to value object
+
+If you want more type safety you can choose to parse the input array into a value object.
+
+```php
+use GraphQL\Type\Definition\Type;
+use GraphQL\Type\Definition\InputObjectType;
+
+final class StoryFiltersInput
+{
+    public string $author;
+    public ?bool $popular;
+    public array $tags;
+
+    public function __construct(string $author, ?bool $popular, array $tags)
+    {
+        $this->author = $author;
+        $this->popular = $popular;
+        $this->tag = $tag;
+    }
+}
+
+$filters = new InputObjectType([
+    'name' => 'StoryFiltersInput',
+    'fields' => [
+        'author' => [
+            'type' => Type::nonNull(Type::string()),
+        ],
+        'popular' => [
+            'type' => Type::boolean(),
+        ],
+        'tags' => [
+            'type' => Type::listOf(Type::string()),
+        ]
+    ],
+    'parseValue' => fn(array $values) => new StoryFiltersInput(
+        $values['author'],
+        $values['popular'],
+        $values['tags'] ?? []
+    ),
+]);
+```
+
+The value of `$args['filters']` will now be an instance of `StoryFiltersInput`.
```

Comment:
Perhaps we can mention how nested inputs are handled. How about the following?

The incoming values are converted using a depth-first traversal. Thus, nested input values
will be passed through their respective `parseValue` functions before the parent receives their value.